airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1334,6 +1336,8 @@ def _merge_stream_slicers(
                 if model.incremental_sync
                 else None
             )
+        elif self._disable_resumable_full_refresh:
+            return stream_slicer
```

Comment:
This was an interesting trick that took me a moment to find.

Right now since for low-code we automatically instantiate an RFR cursor. This was done to automatically release RFR to low-code connectors. However, since we don't want to support RFR for concurrent, we need a flag to signal not to apply it. And returning here works because we either return the `stream_slicer` if it exists or `None`, skipping over auto-creating the RFR cursor.

There is a very specific gotcha that I had originally gotten tripped up on, where we lost the state of of the PerPartitionCursor while instantiating child cursor during partition generation. And when we actually synced the partition, all the child cursors were lost. @maxi297, just tagging you to remind me to mention this in more detail to you, because its guaranteed to come up when incremental substreams is going to be worked on. But right now, this issue doesn't affect substreams that are full refresh

Reply:
That makes sense to me. I'm even wondering: is there a reason with we keep the following condition?
```
        elif stream_slicer:
            # For the Full-Refresh sub-streams, we use the nested `ChildPartitionResumableFullRefreshCursor`
            return PerPartitionCursor(
                cursor_factory=CursorFactory(
                    create_function=partial(ChildPartitionResumableFullRefreshCursor, {})
                ),
                partition_router=stream_slicer,
            )
```
My understanding is that only sources with that inherits directly from `ManifestDeclarativeSource` would keep the RFR behavior, is that correct? Once we roll out this change, it would only be the Connector Builder, right?

Reply:
hmmm that's a good point, i'd thought to leave it in just in case we still wanted to keep RFR in for manifest sources (connector builder sources, or any that inherit ManifestDeclarativeSource instead of ConcurrentSource) that weren't ported to concurrent yet, but still wanted to continue to have RFR.

If we want to be more aggressive and just drop it entirely moving from 6.2.0+ onward, I can drop the entire conditional block and `self_disable_resumable_full_refresh` field, but my thinking is we can leave it in for now and remove it easily later

Reply:
Let's leave it for now and plan for the removal of RFR in the declarative framework so that we do it right. I've added a row in the tech spec for now